demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c

```diff
@@ -1365,19 +1343,29 @@ static OtaMqttStatus_t mqttPublish( const char * const pacTopic,
         LogError( ( "Failed to acquire mqtt mutex for executing MQTT_Publish"
                     ",errno=%s",
                     strerror( errno ) ) );
-    }
-
-    if( mqttStatus != MQTTSuccess )
-    {
-        LogError( ( "Failed to send PUBLISH packet to broker with error = %u.", mqttStatus ) );
 
         otaRet = OtaMqttPublishFailed;
     }
-    else
+
+    if( ( mqttStatus == MQTTSuccess ) && ( qos == 1 ) )
     {
-        LogInfo( ( "Sent PUBLISH packet to broker %.*s to broker.\n\n",
-                   topicLen,
-                   pacTopic ) );
+        /* Calculate relative interval as current time plus number of seconds. */
+        clock_gettime( CLOCK_REALTIME, &ts );
+        ts.tv_sec += MQTT_ACK_TIMEOUT_MS;
+
+        while( ( ret = sem_timedwait( &ackSemaphore, &ts ) ) == -1 && errno == EINTR )
+        {
+            continue;
+        }
+
+        if( ret == -1 )
+        {
```

Comment:
Shouldn't we retry sending the status update if the publish fails?
If we return `OtaMqttPublishFailed`, `processDataHandler` would return an error and the job would fail?

Reply:
Failure to get an ack when it is QoS 1 is also considered publish failure.